react/features/base/tracks/middleware.js

```diff
@@ -135,7 +135,12 @@ MiddlewareRegistry.register(store => next => action => {
             const isVideoTrack = jitsiTrack.type !== MEDIA_TYPE.AUDIO;
 
             if (isVideoTrack) {
-                if (jitsiTrack.isLocal()) {
+                if (jitsiTrack.type === MEDIA_TYPE.PRESENTER) {
+                    APP.conference.mutePresenter(muted);
+                }
+
+                // Make sure we change the video mute state only for camera tracks.
+                if (jitsiTrack.isLocal() && jitsiTrack.videoType !== 'desktop') {
```

Comment:
If the goal is to change only for camera tracks then maybe the condition should be if `videoType === 'camera'` ?
Also with the change now the logic will go to line 146 for local track which is meant for the remote participants. That seems incorrect.

Reply:
I know this looks confusing but the here is what it is intended to do:
1. For the local camera tracks, regular video and presenter, we need to call UI.setVideoMuted() which updates the thumbnail view and notify external API of change in mute state (called inside conference.setVideoMuteStatus()).
2. For the remote tracks (camera+desktop) and for local desktop track, we need to update the thumbnail view so we need to call UI.setVideoMuted().
We don't support a secondary camera source now but if we were to add a new videoType for a secondary camera, then checking for `videoType !== 'desktop'` would still work. I could change it though :)

Reply:
Ok, thanks for the explanation!